inference-engine/src/legacy_api/src/net_pass.cpp

```diff
@@ -1401,6 +1401,23 @@ void convertLayerPrecision(const CNNLayerPtr& layer) {
     }
 }
 
+template <typename NET>
+bool isOutput(NET& net, CNNLayerPtr layer) {
+    return false;
+}
+
+bool isOutput(InferenceEngine::ICNNNetwork& net, CNNLayerPtr layer) {
```

Comment:
This function doesn't make function template specialization. You need to add `template <>` before this functions.